### DISCUSSION TITLE
Advanced Multi Axis Spider Chart

### DISCUSSION THREAD

**Discussion by Ringomed** [has2k1/plotnine#849] [2024 Plotnine Contest]
# 🕸️ Multi Axis Spider Chart

## 🪶 Authors

Zvonimir Boban
[LinkedIn](https://www.linkedin.com/in/zvonimir-boban) | [GitHub](https://github.com/Ringomed)

## 💡 Motivation 

As a long-time R user who ocasionally does work in Python as well, hearing of the plotnine library trying to bring the ggplot experience to Python instantly piqued my curiosity and this competition was the perfect opportunity to try it out.

Inspired by R's ggplot2, the plotnine library is also based on the concept of grammar of graphics, allowing for creation of graphs by stacking multiple layers on top of one another. This powerful concept lets us create essentially any visualization, as long as we know how to code it. Here, I used it to construct an advanced version of a spider chart from scratch. The chart presents a comparison of Titanic passenger across the three passenger classes.

## 📈 The Chart

<p align="center">
<img src="https://github.com/user-attachments/assets/a7c02222-581f-4b49-96d7-4f4b4bcb7ee2">
</p>

Aside from enabling the usual comparison of values on a relative scale, this version of the spider chart also allows a quick insight into the absolute feature values as well, without having to go back to the original dataset. We can clearly see that the 1st class passengers were the oldest and the wealthiest of the three. The 3rd class passengers had the highest number of both male and female passengers and were the youngest group — probably mostly young people and families in search for better life abroad. However, the 1st class passengers had the highest survival rate of 63 %, and the 3rd the lowest (just 24 %). This is probably partly due to the 1st class quarters being closer to the boat deck and partly due to the higher proportion of women in that class (since woman and children were rescued first).

## ➕ Bonus Chart

Out of curiosity, I also tested the chart on the mtcars data to compare the properties of thre different cars. Enjoy!

<p align="center">
<img src="https://github.com/user-attachments/assets/ddd0025e-b936-488a-a356-0a7554e0401c">
</p>

## 💻 Code Repository

You can find all the neccessary files and notebooks with the code at the accompanying [GitHub repository](https://github.com/Ringomed/Plotnine-contest/tree/main). The tutorial detailing each step of chart creation can be found in the [README](https://github.com/Ringomed/Plotnine-contest/blob/main/README.md) file 🙂.

**Comment by has2k1** [comment]
![](https://private-user-images.githubusercontent.com/60142390/351651377-a7c02222-581f-4b49-96d7-4f4b4bcb7ee2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MzUyNzksIm5iZiI6MTcyNDkzNDk3OSwicGF0aCI6Ii82MDE0MjM5MC8zNTE2NTEzNzctYTdjMDIyMjItNTgxZi00YjQ5LTk2ZDctNGY0YjRiY2I3ZWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDEyMzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2NWNiOTJmMzJjNDAxMzI4YWRlY2I4YzAzZTdiZTA3YjBlNzcwN2I4NjYxMTU3YmQ3NGM0NmJhMTYyMzE4MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vp98rbbF0USnhRz3ViPgpqykijme2H40iM6ubyRoeXo)

<hr/>

![](https://private-user-images.githubusercontent.com/60142390/351655316-ddd0025e-b936-488a-a356-0a7554e0401c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MzUyNzksIm5iZiI6MTcyNDkzNDk3OSwicGF0aCI6Ii82MDE0MjM5MC8zNTE2NTUzMTYtZGRkMDAyNWUtYjkzNi00ODhhLWEzNTYtMGE3NTU0ZTA0MDFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDEyMzYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViYTU5ZTFkNWE5OTdkZjA4YjlhN2Q3NTJkNzg4NDg1Nzc5MTQ4YzliOWNhZDliNGM5ZWFiNjg5MWM2NmIyNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ssBFe6NbWmokg7a60VOan5PQBUfWIjnUZZLCYZ7t4qg)

**Reply by Ringomed** [reply]
Hi, from other posts in the discussion thread, I see that you usually created a comment with the plot. I can't see anything in the comment, just an empty post. Was everything ok with the plot?

**Comment by EconomiCurtis** [comment]
Ahoy @Ringomed , 
I am one of the contest organizers. 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Comment by has2k1** [comment]
@Ringomed, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.
